Alerting: Add external Alertmanagers (#39183)
grafana/grafana: docs/sources/alerting/unified-alerting/fundamentals/alertmanager.md

Grafana 8 alerting added support for external Alertmanager configuration. When you add an [Alertmanager data source]({{< relref "../../../datasources/alertmanager.md" >}}), the Alertmanager drop-down shows a list of available external Alertmanager data sources. Select a data source to create and manage alerting for standalone Cortex or Loki data sources.

{{< figure max-width="40%" src="/static/img/docs/alerting/unified/contact-points-select-am-8-0.gif" max-width="250px" caption="Select Alertmanager" >}}

## External Alertmanagers

> It is possible to add one or many external alertmanagers to your alertmanager configuration. This will route your alerts to this/these alertmanager(s) as well as the internal one (??). This is done via the admin tab in the alerting ui.

Author: @gotjosh could you take a look at this section please?

> It is possible to add one or many external alertmanagers to your alertmanager configuration. This will route your alerts to this/these alertmanager(s) as well as the internal one (??). This is done via the admin tab in the alerting ui.

Reviewer: ```suggestion
You can configure one or many external Alertmanagers to receive alerts generated by Grafana. If configured, both the internal Grafana Alertmanager **and** any configured external Alertmanagers will receive _all_ alerts generated by Grafana.

You can do the setup in the "Admin" tab within the Grafana v8 Alerts UI.
```

### Add a new external Alertmanager

1. In the Grafana menu, click the Alerting (bell) icon to open the Alerting page listing existing alerts.

> 2. Click Admin

Reviewer: ```suggestion
2. Click Admin from the set of tabs
```

3. Scroll down to the External Alertmanager section
4. Click the Add Alertmanager button, a modal opens.

> 5. Add the url and eventual port to the alertmanager, there is no need to specify`/api/v(1|2)/alerts`. Grafana will add this.

Reviewer: ```suggestion
5. Add the URL and port for the Alertmanager you wish to configure, there is no need to specify the path suffix e.g. `/api/v(1|2)/alerts`. Grafana will add this.
```

> 6. The url will be shown in the table with a pending status, after a short while when Grafana has verified/connected/resolved it will change to active.

Reviewer: ```suggestion
6. The URL is shown in the table with a pending status, after a short period Grafana will verify that the Alertmanager is discovered and it will change to active. No requests are made to the external Alertmanager at this point, the verification is meant to signal that we're ready to send but not that sends will be successful. 
```

### Edit an external Alertmanager

> 1. Click the pen symbol to the right of the alertmanager row in the table.

Reviewer: ```suggestion
1. Click the pen symbol to the right of the Alertmanager row in the table.
```

> 2. The modal will open, and you will see all urls added. Change the url you want to update and click the Add Alertmanager button.

Reviewer: ```suggestion
2. When the modal opens, you will see all URLs added. Change the URL you want to update and click the Add Alertmanager button.
```

> 3. The edited url will be pending until Grafana has verified/connected/resolved.

Reviewer: ```suggestion
3. The edited URL will be pending until Grafana verifies it again.
```

Grafana 8 alerting added support for external Alertmanager configuration. When you add an [Alertmanager data source]({{< relref "../../../datasources/alertmanager.md" >}}), the Alertmanager drop-down shows a list of available external Alertmanager data sources. Select a data source to create and manage alerting for standalone Cortex or Loki data sources.

{{< figure max-width="40%" src="/static/img/docs/alerting/unified/contact-points-select-am-8-0.gif" max-width="250px" caption="Select Alertmanager" >}}

> ## External Alertmanagers

Reviewer 2: ```suggestion
```

> You can configure one or many external Alertmanagers to receive alerts generated by Grafana. If configured, both the internal Grafana Alertmanager **and** any configured external Alertmanagers will receive _all_ alerts generated by Grafana.
> 
> You can do the setup in the "Admin" tab within the Grafana v8 Alerts UI.

Reviewer 2: ```suggestion
You can configure one or several external Alertmanagers to receive alerts from Grafana. Once configured, both the embedded Alertmanager **and** any configured external Alertmanagers will receive _all_ alerts.
```

### Add a new external Alertmanager

1. In the Grafana menu, click the Alerting (bell) icon to open the Alerting page listing existing alerts.

> 2. Click Admin from the set of tabs
> 3. Scroll down to the External Alertmanager section

Reviewer 2: ```suggestion
2. Click **Admin** and then scroll down to the External Alertmanager section.
```

> 4. Click the Add Alertmanager button, a modal opens.

Reviewer 2: ```suggestion
4. Click **Add Alertmanager** and a modal opens.
```

> 5. Add the URL and port for the Alertmanager you wish to configure, there is no need to specify the path suffix e.g. `/api/v(1|2)/alerts`. Grafana will add this.

Reviewer 2: ```suggestion
5. Add the URL and the port for the external Alertmanager. You do not need to specify the path suffix, for example,  `/api/v(1|2)/alerts`. Grafana automatically adds this.
```

> 6. The URL is shown in the table with a pending status, after a short period Grafana will verify that the Alertmanager is discovered and it will change to active. No requests are made to the external Alertmanager at this point, the verification is meant to signal that we're ready to send but not that sends will be successful.

Reviewer 2: Not a step.

> 6. The URL is shown in the table with a pending status, after a short period Grafana will verify that the Alertmanager is discovered and it will change to active. No requests are made to the external Alertmanager at this point, the verification is meant to signal that we're ready to send but not that sends will be successful.

Reviewer 2: ```suggestion
The external URL is listed in the table with a pending status. Once Grafana verifies that the Alertmanager is discovered, the status changes to active. No requests are made to the external Alertmanager at this point; the verification signals that alerts are ready to be sent.
```

### Edit an external Alertmanager

1. Click the pen symbol to the right of the Alertmanager row in the table.

> 2. When the modal opens, you will see all URLs added. Change the URL you want to update and click the Add Alertmanager button.

Reviewer 2: ```suggestion
1. When the edit modal opens, you can view all the URLs that were added. 
1. Change the URL you want to update.
1. Click **Add Alertmanager** to save your changes.
```

> 3. The edited URL will be pending until Grafana verifies it again.

Reviewer 2: ```suggestion
The updated URL will be in a pending state until Grafana verifies it again.
```
